sql/stringtype.go

```diff
@@ -204,12 +204,43 @@ func (t stringType) Convert(v interface{}) (interface{}, error) {
 		return nil, nil
 	}
 
-	if ti, ok := v.(time.Time); ok {
-		v = ti.Format(TimestampDatetimeLayout)
-	}
-
-	val, err := cast.ToStringE(v)
-	if err != nil {
+	var val string
+	switch s := v.(type) {
+	case bool:
+		val = strconv.FormatBool(s)
+	case float64:
+		val = strconv.FormatFloat(s, 'f', -1, 64)
+	case float32:
+		val = strconv.FormatFloat(float64(s), 'f', -1, 32)
+	case int:
+		val = strconv.FormatInt(int64(s), 10)
+	case int8:
+		val = strconv.FormatInt(int64(s), 10)
+	case int16:
+		val = strconv.FormatInt(int64(s), 10)
+	case int32:
+		val = strconv.FormatInt(int64(s), 10)
+	case int64:
+		val = strconv.FormatInt(s, 10)
+	case uint:
+		val = strconv.FormatUint(uint64(s), 10)
+	case uint8:
+		val = strconv.FormatUint(uint64(s), 10)
+	case uint16:
+		val = strconv.FormatUint(uint64(s), 10)
+	case uint32:
+		val = strconv.FormatUint(uint64(s), 10)
+	case uint64:
+		val = strconv.FormatUint(s, 10)
+	case string:
+		val = s
+	case []byte:
+		val = string(s)
+	case time.Time:
+		val = s.Format(TimestampDatetimeLayout)
+	case fmt.Stringer:
```

Comment:
Might not want this in here, pretty general. What does this catch that the above cases don't?

Reply:
It was in the previous `cast` package implementation, so I carried it over so that anything that relied on it before continued to work. I can remove it though.

Reply:
Yank it out and see what breaks. Feel like we want every type to be handled explicitly by this block, not by the type itself.